src/helm/proxy/clients/google_client.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Dict
+
+from helm.common.cache import Cache, CacheConfig
+from helm.common.request import Request, RequestResult, Sequence, Token
+from helm.common.tokenization_request import (
+    TokenizationRequest,
+    TokenizationRequestResult,
+    DecodeRequest,
+    DecodeRequestResult,
+)
+from .client import Client, truncate_sequence
+
+
+class GoogleClient(Client):
+    """
+    Client for the Google models. There isn't an API for their language models.
+    We receive and process completions offline.
+    """
+
+    @staticmethod
+    def convert_to_raw_request(request: Request) -> Dict:
+        return {
+            "best_of": request.top_k_per_token,
+            "echo": request.echo_prompt,
+            "logprobs": request.top_k_per_token,
+            "max_tokens": request.max_tokens,
+            "model": request.model_engine,
+            "n": request.num_completions,
+            "prompt": request.prompt,
+            "request_type": "language-model-inference",
+            "stop": request.stop_sequences or None,
+            "temperature": request.temperature,
+            "top_p": request.top_p,
+        }
+
+    def __init__(self, cache_config: CacheConfig):
+        self.cache = Cache(cache_config)
+
+    def make_request(self, request: Request) -> RequestResult:
+        raw_request = GoogleClient.convert_to_raw_request(request)
+        cache_key: Dict = Client.make_cache_key(raw_request, request)
+
+        try:
+
+            def fail():
+                raise RuntimeError(
+                    f"The result has not been uploaded to the cache for the following request: {cache_key}"
+                )
+
+            # If results are not cached for a given query, fail fast
+            response, cached = self.cache.get(cache_key, fail)
+        except RuntimeError as e:
+            error: str = f"GoogleClient error: {e}"
+            return RequestResult(success=False, cached=False, error=error, completions=[], embedding=[])
+
+        # Expect the result to be structured the same way as a response from OpenAI API.
+        completions: List[Sequence] = []
+        for raw_completion in response["choices"]:
```

Comment:
Question: Is the response schema documented somewhere?

Reply:
It's pretty much structured the same way as OpenAI's response (see comment above).